### PR TITLE
Tools: Topology2: Add widget class IGO NR

### PR DESCRIPTION
**Pull request by singalsu**

### CODE REVIEW

**Review by lgirdwood — APPROVED** [review]
@ranj063 @jsarha

File: tools/topology/topology2/include/components/igo_nr.conf [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#
+#
+# A generic IGO_NR component. All attributes defined herein are namespaced
```
Comment (btian1) [inline tools/topology/topology2/include/components/igo_nr.conf:3]: I suggest add full name of IGO_NR in a place, let everyone know what exactly IGO_NR is.

File: tools/topology/topology2/include/components/igo_nr.conf [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                        Object.Base.channel.1 {
+                                name    "fl"
+                                shift	0
+                        }
+                        Object.Base.channel.2 {
+                                name    "fr"
+                        }
+                        Object.Base.ops.1 {
+                                name	"ctl"
+                                info	"volsw"
```
Comment (btian1) [inline tools/topology/topology2/include/components/igo_nr.conf:73]: add comment for what "volsw" is, for all abbreviation names that not so obvious to know the meanning.

File: tools/topology/topology2/cavs-benchmark-hda.conf [hunk 3]
```diff
@@ -284,4 +285,20 @@ IncludeByKey.BENCH_CONFIG {
...
+		<include/bench/igo_nr_s16.conf>
+	}
+
+	"igo_nr24" {
+		<include/bench/igo_nr_s24.conf>
+	}
+
+	"igo_nr32" {
+		<include/bench/igo_nr_s32.conf>
+	}
```
Comment (btian1) [inline tools/topology/topology2/cavs-benchmark-hda.conf:303]: possible combine and use one conf for 16/24/32? all three files are short and similar. @ranj063 @singalsu

Comment (singalsu) [reply tools/topology/topology2/cavs-benchmark-hda.conf:303]: This is automatically generated "code" and if some format is not supported by a component, I just leave out from commit the unused files e.g. aria. In this case I prefer the small test topology .conf building blocks. If a change is I change the generator bench_comp_generate.sh and re-run it for the component.

File: tools/topology/topology2/include/components/igo_nr.conf (hunk 2 above)
Comment (singalsu) [reply tools/topology/topology2/include/components/igo_nr.conf:73]: @ranj063 @jsarha Do you know where this info is used? This has been "volsw" in all switch controls I've seen, so I've just repeated it. The control name for user space is defined elsewhere. Does the user see somehow this info text?

File: tools/topology/topology2/include/components/igo_nr.conf (hunk 1 above)
Comment (btian1) [reply tools/topology/topology2/include/components/igo_nr.conf:3]: @singalsu , are you adding full name for IGO_NR?

File: tools/topology/topology2/include/components/igo_nr.conf (hunk 2 above)
Comment (jsarha) [reply tools/topology/topology2/include/components/igo_nr.conf:73]: The legal values are "volsw", "volsw_sx" and "volsw_xr_sx", and the value describes the kcontrol type of the mixer. It somehow describes how the mixer values are interpreted. The relevant functions are implemented in sound/soc/soc-ops.c, but I can not immediately say what is the relevant difference, or find any piece of documentation describing it.

Comment (singalsu) [reply tools/topology/topology2/include/components/igo_nr.conf:73]: Thanks @jsarha I'll check if I can understand what they are and add a comment. This has not been clear to me either, I've just copied working topology examples from others.

File: tools/topology/topology2/include/components/igo_nr.conf (hunk 1 above)
Comment (singalsu) [reply tools/topology/topology2/include/components/igo_nr.conf:3]: Yep, I will update this simple PR next. Don't approve yet.

File: tools/topology/topology2/include/components/igo_nr.conf (hunk 2 above)
Comment (singalsu) [reply tools/topology/topology2/include/components/igo_nr.conf:73]: Volsw is the single or double mixer control. The _sx version is for use where signed values can be both sides of zero. _xr_sx is some multi-register to single combine control. I'll add some comment but it's not going to be well descriptive. E.g. ALSA documentation is not describing these, only lists the operations names @jsarha mentioned above.

**Comment by lgirdwood** [comment]
@btian1 @jsarha good for you now, if so pls approve.

**Review by jsarha — APPROVED** [review]

**Pull request merged**